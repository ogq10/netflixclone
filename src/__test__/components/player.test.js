import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Player from "../../components/player";

describe("<Player/>", () => {
  it("renders the Player component with a warner bros intro video", () => {
    const { container, getByText, queryByTestId, debug } = render(
      <Player>
        <Player.Button />
        <Player.Video src="/videos/warner.mp4" />
      </Player>
    );

    expect(queryByTestId("player")).toBeFalsy();
    fireEvent.click(getByText("Play"));

    expect(queryByTestId("player")).toBeTruthy();
    fireEvent.click(queryByTestId("player"));

    expect(queryByTestId("player")).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
