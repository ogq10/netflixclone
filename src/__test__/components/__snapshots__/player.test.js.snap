// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player/> renders the Player component with a warner bros intro video 1`] = `
<div
  class="player__Container-sc-16pqj6x-0"
>
  <button
    class="player__Button-sc-16pqj6x-3 evQiap"
  >
    Play
  </button>
   
</div>
`;
